### ISSUE TITLE
Video playback is not smooth on Mi Box S

### ISSUE THREAD

**Issue by alexeyr15** [androidx/media#1084]
### Version

Media3 1.2.1

### Devices that reproduce the issue

Mi Box S

### Devices that do not reproduce the issue

Nvidia Shield, Google Chromecast

### Reproducible in the demo app?

Yes

### Reproduction steps

1. Play the test video stream in the demo app on Mi Box S. You will need to enable usesCleartextTraffic, FLAG_ALLOW_NON_IDR_KEYFRAMES and ffmpeg extension.
2. Video playback is not smooth, stuttering.

I was able to reproduce this issue on Mi Box S only. It is not reproduced on Mi Box S with the old versions of ExoPlayer like 2.11.0.

### Expected result

Video playback should be smooth

### Actual result

Video playback is not smooth

### Media

The link to the test video stream will be sent by email

### Bug Report

- [ ] You will email the zip file produced by `adb bugreport` to android-media-github@google.com after filing this issue.

**Comment by marcbaechinger** [comment]
Thanks for your report. 

> You will need to enable [..] ffmpeg extension

From your report 'Video playback is not smooth', I understand that you see problems with video decoding/rendering. 

The ffmpeg extension you mention is as far as I can tell audio-only as the README says: "The FFmpeg module provides FfmpegAudioRenderer, which uses FFmpeg for decoding and can render audio encoded in a variety of formats."

Can you clarify a bit how you play the stream and why ffmpeg is required to play it? If you'd provide a bit more details why your case is special and needs ffmpeg, this would speed analysis up on our side.

I understand the audio track can't be played without ffmpeg. What is the behaviour when you disable the audio track? Do you still see problems with rendering the video?

I don't have a MI Box S at hand right now. Can you please repro the problem and then do a bug report just after the problem exhibited? The please upload the bug report here or sent it to email address you've sent the stream link to. Please let us know you've done this here also. We don't check the inbox of the support email address without request here.

**Comment by alexeyr15** [comment]
> I understand the audio track can't be played without ffmpeg.

Yes.

> What is the behaviour when you disable the audio track? Do you still see problems with rendering the video?

I still see problems with rendering the video. Also, the video playback is not smooth in the demo app built without ffmpeg audio extension.

> Can you please repro the problem and then do a bug report just after the problem exhibited? 

I have sent the bug report by email.

**Comment by marcbaechinger** [comment]
> the video playback is not smooth in the demo app built without ffmpeg

Yes. It doesn't at all play for me. Ideas? (it's ok to get more verbose)

**Comment by alexeyr15** [comment]
> Yes. It doesn't at all play for me. Ideas? (it's ok to get more verbose)

Try to add FLAG_ALLOW_NON_IDR_KEYFRAMES flag:

```
  public DefaultHlsExtractorFactory() {
    this(/* payloadReaderFactoryFlags= */ DefaultTsPayloadReaderFactory.FLAG_ALLOW_NON_IDR_KEYFRAMES, /* exposeCea608WhenMissingDeclarations */ true);
//    this(/* payloadReaderFactoryFlags= */ 0, /* exposeCea608WhenMissingDeclarations */ true);
  }
```

**Comment by alexeyr15** [comment]
Have you managed to reproduce this issue?

**Comment by alexeyr15** [comment]
Have you had a chance to check this on Mi Box S?
